python/cudf/tests/test_dataframe.py

```diff
@@ -2169,6 +2169,69 @@ def test_reset_index(pdf, gdf, drop):
               gdf.x.reset_index(drop=drop))
 
 
+@pytest.mark.parametrize('copy', [True, False])
+def test_reindex(copy):
+    index = [-3, -2, 0, 1, 3, 4, 6]
```

Comment:
Should add a case where there's duplicates in here (should already be handled by your code).